internal/dag/secret.go

```diff
@@ -59,9 +60,9 @@ func validateCertificate(data []byte) error {
 }
 
 func validatePrivateKey(data []byte) error {
-	key, err := decodePEM(data)
+	key, err := decodePEMPrivateKey(data)
 	if err != nil {
-		return nil
+		return err
```

Comment:
I wasn't sure why err was not previously returned (failure to parse the PEM should be an error?)